### PR TITLE
[OCC] refine code, use switch instead of if

### PR DESCRIPTION
**Pull request by wolfkdy**

### CODE REVIEW

File: utilities/transactions/optimistic_transaction.cc [hunk 1]
```diff
@@ -60,12 +60,13 @@ Status OptimisticTransaction::Commit() {
                                             OptimisticTransactionDB>(txn_db_);
   assert(txn_db_impl);
   OccValidationPolicy policy = txn_db_impl->GetValidatePolicy();
-  if (policy == OccValidationPolicy::kValidateParallel) {
-    return CommitWithParallelValidate();
-  } else if (policy == OccValidationPolicy::kValidateSerial) {
-    return CommitWithSerialValidate();
-  } else {
-    assert(0);
+  switch (policy) {
```
Comment (siying) [inline utilities/transactions/optimistic_transaction.cc]: Minor: you can move `txn_db_impl->GetValidatePolicy()` to here too.

**Comment by zhichao-cao** [comment]
@wolfkdy Can you rebase this diff? It will be landed after rebase.

**Comment by wolfkdy** [comment]
@zhichao-cao  sorry for the delayed reply, this will be done today.

File: utilities/transactions/optimistic_transaction.cc (hunk 1 above)
Comment (wolfkdy) [reply utilities/transactions/optimistic_transaction.cc]: done

**Comment by zhichao-cao** [comment]
You may use a wrong git commend to rebase, which causes such a large number of change. You may need to roll back, do git fetch upstream, git rebase upstream/master.

**Comment by wolfkdy** [comment]
@zhichao-cao  , how about now? 
```
DEYUKONG-MB0:rocksdb kdy$ git remote -v
origin	https://github.com/wolfkdy/rocksdb.git (fetch)
origin	https://github.com/wolfkdy/rocksdb.git (push)
upstream	https://github.com/facebook/rocksdb.git (fetch)
upstream	https://github.com/facebook/rocksdb.git (push)
DEYUKONG-MB0:rocksdb kdy$ git fetch upstream
DEYUKONG-MB0:rocksdb kdy$ git rebase upstream/master
Current branch wolf_parallel_occ6 is up to date.
DEYUKONG-MB0:rocksdb kdy$ git push
Everything up-to-date
DEYUKONG-MB0:rocksdb kdy$ 

```

**Comment by zhichao-cao** [comment]
> @zhichao-cao , how about now?
> 
> ```
> DEYUKONG-MB0:rocksdb kdy$ git remote -v
> origin	https://github.com/wolfkdy/rocksdb.git (fetch)
> origin	https://github.com/wolfkdy/rocksdb.git (push)
> upstream	https://github.com/facebook/rocksdb.git (fetch)
> upstream	https://github.com/facebook/rocksdb.git (push)
> DEYUKONG-MB0:rocksdb kdy$ git fetch upstream
> DEYUKONG-MB0:rocksdb kdy$ git rebase upstream/master
> Current branch wolf_parallel_occ6 is up to date.
> DEYUKONG-MB0:rocksdb kdy$ git push
> Everything up-to-date
> DEYUKONG-MB0:rocksdb kdy$ 
> ```

LGTM. Thank you!

**Comment by zhichao-cao** [comment]
@wolfkdy Can you rebase again? Landing is rejected. After rebase and push, I will land again for you.

**Comment by wolfkdy** [comment]
@zhichao-cao  I create a new branch from the branch new upsteram/master and "git merge --squash".
Maybe it will fix everything.  https://github.com/facebook/rocksdb/pull/6420

**Comment by zhichao-cao** [comment]
Changes moved to #6420 , close this one.

**Pull request closed without merging**